modules/scripts/dtbook-cleaner/src/main/resources/xml/dtbook-cleaner.script.xpl

```diff
@@ -0,0 +1,164 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<p:declare-step xmlns:p="http://www.w3.org/ns/xproc" version="1.0"
+                xmlns:px="http://www.daisy.org/ns/pipeline/xproc"
+                xmlns:pxi="http://www.daisy.org/ns/pipeline/xproc/internal"
+                xmlns:d="http://www.daisy.org/ns/pipeline/data"
+                name="main"
+                type="px:dtbook-cleaner.script"
+                px:input-filesets="dtbook"
+                px:output-filesets="dtbook"
+                exclude-inline-prefixes="#all">
+
+    <p:documentation xmlns="http://www.w3.org/1999/xhtml">
+        <h1 px:role="name">DTBook Cleaner</h1>
+        <p px:role="desc">Cleaning DTBook documents</p>
+        <div px:role="author maintainer">
+            <p px:role="name">Nicolas Pavie</p>
+            <a px:role="contact" href="mailto:pavie.nicolas@gmail.com">pavie.nicolas@gmail.com</a>
+            <p px:role="organization">DAISY Consortium</p>
+        </div>
+    </p:documentation>
+
+    <!-- ***************************************************** -->
+    <!-- IMPORTS -->
+    <!-- ***************************************************** -->
+    <p:import href="http://www.daisy.org/pipeline/modules/dtbook-utils/library.xpl">
+        <p:documentation>
+            px:dtbook-fix
+        </p:documentation>
+    </p:import>
+    <p:import href="http://www.daisy.org/pipeline/modules/dtbook-break-detection/library.xpl">
+        <p:documentation>
+            px:dtbook-break-detect
+            px:dtbook-unwrap-words
+        </p:documentation>
+    </p:import>
+
+    <!-- ***************************************************** -->
+    <!-- INPUTS / OUTPUTS / OPTIONS -->
+    <!-- ***************************************************** -->
+
+    <p:input port="source" primary="true" sequence="true" px:media-type="application/x-dtbook+xml">
+        <p:documentation xmlns="http://www.w3.org/1999/xhtml">
+            <h2 px:role="name">DTBook file(s)</h2>
+            <p px:role="desc">One or more DTBook files to be cleaned</p>
+        </p:documentation>
+    </p:input>
+
+    <p:output port="result" primary="true" sequence="true" px:media-type="application/x-dtbook+xml">
+        <p:documentation xmlns="http://www.w3.org/1999/xhtml">
+            <h2 px:role="name">DTBook file(s)</h2>
+            <p px:role="desc">Cleaned DTBooks</p>
+        </p:documentation>
+        <p:pipe step="cleaned-dtbook" port="result"/>
+    </p:output>
+
+
+    <p:option name="output-dir" required="true" px:output="result" px:type="anyDirURI">
+        <p:documentation xmlns="http://www.w3.org/1999/xhtml">
+            <h2 px:role="name">DTBook</h2>
+            <p px:role="desc">The directory that will contain the resulting DTBook.</p>
+        </p:documentation>
+    </p:option>
+
+
+    <!-- Options from dtbook-fix script -->
+    <p:option name="repair" required="false" select="false()" px:type="boolean">
```

Comment:
Script option are always of type string. So the `select` attribute must become: `select="'false'"`. The `px:type="boolean"` is for telling the user interface what are the allowed values.

Note that this also means that in the `px:dtbook-fix` call below, in order to pass the options as booleans, the `p:with-option` must become: `<p:with-option name="repair" select="$repair='true'"/>`

Reply:
Fixed